master/test/streamdata/stream_data.go

```diff
@@ -66,55 +68,31 @@ type trial struct {
 
 // ModTrial is a convenience function for modifying rows of a trial.
 func ModTrial(
-	ctx context.Context, trialID, experimentID int, changeStart bool,
-	changeSeq bool, changeState string,
+	ctx context.Context, newTrial Trial,
 ) error {
 	trials, err := queryTrials(ctx)
 	if err != nil {
 		return err
 	}
-	startTime := time.Time{}
-	seq := int64(0)
 	maxSeq := int64(0)
-	state := model.State("")
 	for _, t := range trials {
-		if t.ID == trialID {
-			startTime = t.StartTime
-			seq = t.Seq
-			state = t.State
-		}
 		if t.Seq > maxSeq {
 			maxSeq = t.Seq
 		}
 	}
 
-	if changeStart {
-		startTime = time.Now()
-	}
-	if changeSeq {
-		seq = maxSeq + 1
-	}
-	if changeState != "" {
-		state = model.State(changeState)
-	}
+	newTrial.Seq = maxSeq + 1
```

Comment:
question: why would you update the seq value? Won't it be handled by the trigger? Are you trying to avoid checking if the row was updated with a query here?

Reply:
oops I thought I commented - that's a good point, I thought I had to update it for new trials/experiments but it definitely should not be there for updates

Reply:
I guess I'm unclear on what your getting at in your response?